service/src/main/java/uk/gov/hmcts/reform/fpl/service/OthersService.java

```diff
@@ -39,4 +48,86 @@ private boolean otherExists(Others others) {
         return others != null && (others.getFirstOther() != null || others.getAdditionalOthers() != null);
     }
 
+    public Others modifyHiddenValues(List<Element<Other>> others) {
+        List<Element<Other>> othersList = new ArrayList<>();
+
+        others.forEach(other -> {
+            if (other.getValue().containsConfidentialDetails()) {
+                othersList.add(
+                    element(other.getId(), other.getValue().toBuilder()
+                        .address(null)
+                        .telephone(null)
+                        .build()));
+            } else {
+                othersList.add(other);
+            }
+        });
+
+        return new Others(othersList.get(0).getValue(), othersList.subList(1, othersList.size()));
+    }
+
+    public List<Element<Other>> retainConfidentialDetails(List<Element<Other>> confidentialOthers) {
+        final List<Element<Other>> confidentialOthersModified = new ArrayList<>();
+
+        confidentialOthers.forEach(element -> confidentialOthersModified.add(
+            element(element.getId(), Other.builder()
+                .address(element.getValue().getAddress())
+                .telephone(element.getValue().getTelephone())
+                .build())));
+
+        return confidentialOthersModified;
+    }
+
+    public Others prepareOthers(CaseData caseData) {
+        List<Element<Other>> others = new ArrayList<>();
+
+        caseData.getAllOthers().forEach(element -> {
+            if (element.getValue().containsConfidentialDetails()) {
+                Element<Other> confidentialOther = findConfidentialOther(caseData.getConfidentialOthers(), element);
+
+                others.add(element(element.getId(), addConfidentialDetails(confidentialOther, element)));
+            } else {
+                others.add(element);
+            }
+        });
+
+        Other firstOther = getFirstOther(caseData.getConfidentialOthers(), others);
+
+        if (isNotEmpty(others)) {
+            others.remove(0);
```

Comment:
Why are we removing first element here?

Reply:
so the getAllOthers method will always place the firstOther as element 0. As we want to pull out the firstOther(with confidential details if they're there) I have this in the getFirstOther method. We then need to remove the firstOther from the list so it is not repeated. Agree this isn't clear, any ideas on a better implementation whilst I work through the rest of the comments?